frontend/src/views/Market.vue

```diff
@@ -633,7 +713,7 @@ export default Vue.extend({
 
         this.searchResultsOwned = this.search === this.defaultAccount;
         this.waitingMarketOutcome = false;
-        this.searchResults = result;
+        this.searchResults = await this.filterOutTargetBuyers(result) as string[];
```

Comment:
I assume this cast was added because the type for `filterOutTargetBuyers` was somehow not picked up. Did you try explicitly specifying the return type on the `filterOutTargetBuyers` method itself?

Reply:
Seems to work when I do `: Promise<string>`. I will get rid of the casts.